### DISCUSSION TITLE
Components to right of Upload

### DISCUSSION THREAD

**Discussion by michaelcsikos** [ant-design-blazor/ant-design-blazor#2963] [Q&A]
I have some buttons above a `Table` including an `Upload` with a hidden `InputFile` like [this demo](https://antblazor.com/en-US/components/upload#components-upload-demo-InputFile_). I would like to place more components to the right such as a `Search`. Anything placed after the `Upload` is forced underneath. Is there any way to solve this?

![image](https://user-images.githubusercontent.com/21806535/209232921-4f8ab9c3-e5ad-4d0b-bfff-0b340d72966f.png)

**Comment by ElderJames** [comment]
> Anything placed after the Upload is forced underneath. 

What does that mean, please?

**Comment by michaelcsikos** [comment]
Hi @ElderJames, I would like the `Search` to be inline with the `Upload` like this:

![image](https://user-images.githubusercontent.com/21806535/209918546-cfaaf4ef-08d0-4b6c-adfd-b52b69cb4642.png)

Even better would be to also have the `Search` aligned to the right of the page. However, any component after the `Upload` gets displayed below, not to the right.

**Reply by ElderJames** [reply]
I haven't noticed this problem. I think it is a bug. Are you interested in helping to fix it?

**Reply by michaelcsikos** [reply]
The `Upload` does not use any `<div>` when `!ShowUploadList` but the `UploadButton` is using a `<div>`. Could this be the problem?

**Reply by ElderJames** [reply]
You can try warp the `<label>` with a `<div>`.

```
<InputFile id="fileInput" OnChange="UploadFiles" hidden multiple />

<Upload Name="files" FileList="fileList">
  <div>
    <label class="ant-btn" for="fileInput">
        <Icon Type="upload" />
        Click to Upload
    </label>
  </div>
</Upload>

```

**Reply by michaelcsikos** [reply]
I think the `UploadButton` should use a `<span>` instead of a `<div>`.

`<div>` always pushes its components onto the next line. With the following code:
```razor
<div>
    <p>First 3</p>

    <Search WrapperStyle = "width: 100px;" />
    <span>
        <Search WrapperStyle = "width: 100px;" />
        <Search WrapperStyle = "width: 100px;" />
    </span>
</div>

<br />

<div>
    <p>Second 3</p>

    <Search WrapperStyle = "width: 100px;" />
    <div>
        <Search WrapperStyle = "width: 100px;" />
        <Search WrapperStyle = "width: 100px;" />
    </div>
</div>
```

The `<span>` keeps the next 2 `Search` components in-line, but the `<div>` pushes it down in the _Second 3_.

![image](https://user-images.githubusercontent.com/21806535/210047635-3124385f-f0b0-45ba-92fd-4f991fa3bef1.png)

**Reply by ElderJames** [reply]
That's fine. You can submit a PR for that.
